app/controllers/reports_controller.rb

```diff
@@ -132,14 +132,18 @@ def set_wip
     @report.wip = params[:commit] == 'WIP'
   end
 
-  def redirect_url(report)
-    report.wip? ? edit_report_url(report) : report
+  def redirect_url(report, params_hash = {})
+    report.wip? ? edit_report_url(report) : report_url(report, **params_hash)
   end
 
   def notice_message(report)
     report.wip? ? '日報をWIPとして保存しました。' : '日報を保存しました。'
   end
 
+  def params_hash(report)
+    { notify_qa: report.wip? ? false : report.sad? }
```

Comment:
三項演算子を使うより、 `&&` で繋げた方が読みやすいかなと思いました。
（ただ、これは個人の好みかも）

```suggestion
    { notify_qa: !report.wip? && report.sad? }
```

Reply:
@sinsoku コメントありがとうございます。おっしゃる通り条件式の方がスッキリしますね。変更しました 😄